seqr/utils/elasticsearch/constants.py

```diff
@@ -172,6 +173,16 @@
     }
 }
 
+
+def _get_ranks_by_gene(families):
+    ranks_by_gene = defaultdict(list)
+    for record in PhenotypePrioritization.objects.filter(individual__family=families[0]):
```

Comment:
In general you really want to do as much database interaction in the database ORM as possible, not query all the data and post-process it. In the case, you can use the following to pull the data out already grouped:
```
family_ranks = PhenotypePrioritization.objects.filter(
  individual__family=families[0], rank__lte=100).values('gene_id').annotate(min_rank= Min('rank'))
return {agg['gene_id']: agg['min_rank] for agg in family_ranks}
```